salt/_modules/metalk8s_kubernetes.py

```diff
@@ -2306,70 +2310,49 @@ def node_set_taints(node_name, taints, **kwargs):
         _cleanup(**cfg)
 
 
-def node_unschedulable(node_name, **kwargs):
-    '''
-    Return unschedulable value of the node identified by the specified name
+# TODO: Attempt for a central point to call kubernets API,
+#       in order to have a centralized and consistent way
+#       of dealing with errors.
+def call_kube_api(api_instance, method_name, *args, **kwargs):
```

Comment:
That's not used in this PR, maybe you could keep it a branch for later use?

Reply:
Pushed in branch `improvement/generic-kube-api-calls`